File: 49.py
Time Complexity : O(n)
Space Complexity : O(n)
Runs on LeetCode : Yes
Doubts :  # Should we include space complexity as O(n) even though we use it to print to output?


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        dict1={}
        
        for i in strs:
            x = str(sorted(i))
            if x not in dict1:
                dict1[x] =[i]
            else:
                dict1[x].append(i)
        return dict1.values()

File: 49.js
// Time Complexity : O(n) : to navigate the array + O(k log k) to sort K length of strings
//Hence it is O(nklogk)
// Space Complexity : O(nk)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : No

/**
Algorithm :
Create a sorted array of the strings.
Now map each of the unique one as key and corresponding index in strs as value

Return the values.
**/

/**
 * @param {string[]} strs
 * @return {string[][]}
 */
const groupAnagrams = strs => {
	const map = {};
	for (let str of strs) {
		
		//sort the strings
		const sorted = str.split('').sort().join('');
		
		//map sorted string as key and given corresponding strig as value
		map[sorted] !== undefined ? map[sorted].push(str) : map[sorted] = [str];
	}
	  return Object.values(map);
  };